src/core/main.c

```diff
@@ -2459,11 +2486,19 @@ int main(int argc, char *argv[]) {
                 }
         }
 
+        /* Save the original RLIMIT_NOFILE/RLIMIT_MEMLOCK so that we can reset it later when
+         * transitioning from the initrd to the main systemd or suchlike. */
+        if (getrlimit(RLIMIT_NOFILE, &saved_rlimit_nofile) < 0)
+                return log_warning_errno(errno, "Reading RLIMIT_NOFILE failed, ignoring: %m");
+
+        if (getrlimit(RLIMIT_MEMLOCK, &saved_rlimit_memlock) < 0)
+                return log_warning_errno(errno, "Reading RLIMIT_MEMLOCK failed, ignoring: %m");
```

Comment:
not that this matters, but maybe split this out into a function of its own? save_rlimits()?

Reply:
this is main() you are exiting from here, and main() doesn't take "-errno" style error codes. Moreover, you log with LOG_WARNING level here, but die. We shouldn't die because of warnings, only for errors. I think this should be downgraded, so that we proceed if this doesn't wotk.

Reply:
oops yeah... let's create save_rlimits() then and proceed if it doesn't work.